src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const currentState = { ...state };
```

Comment:
The name `currentState` can be misleading here since it suggests that this variable holds the current state throughout the function, but it's actually being mutated. Consider renaming it to `stateCopy` or similar to reflect that it's a mutable copy of the state.